python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -89,7 +89,16 @@ def count(self, split_every=None, split_out=1):
         return groupby_agg(
             self.obj,
             self.by,
-            {c: "count" for c in self.obj.columns if c not in self.by},
+            {
+                c: "count"
+                for c in self.obj.columns
+                if c
+                not in (
+                    self.by
+                    if isinstance(self.by, (tuple, list))
+                    else [self.by]
+                )
+            },
```

Comment:
https://docs.dask.org/en/latest/generated/dask.dataframe.DataFrame.groupby.html#dask-dataframe-dataframe-groupby

> Notice that a tuple is interpreted as a (single) key.

I don't think tuples are meant to work the same as lists here. I think a tuple is meant to be treated as a single object rather than a sequence, which would follow the behavior for `str`.

Also we should factor out this logic into a private helper method. It's 10 lines repeated 10 times.
```suggestion
            {
                c: "count"
                for c in self.obj.columns
                if (c in self.by if isinstance(self.by, list) else c == self.by)
            },
```

```python
def _columns_not_in_by(self):
    if isinstance(self.by, list):
        for c in self.obj.columns:
            if c not in self.by:
                yield c
    else:
        for c in self.obj.columns:
            if c != self.by:
                yield c
```

```suggestion
            {
                c: "count"
                for c in self._columns_not_in_by()
            },
```

Maybe even factor out the dict construction to the helper and pass in the aggregation name?

Reply:
Thanks for the helpful suggestions 😄